schemars/tests/integration/jiff.rs

```diff
@@ -0,0 +1,43 @@
+use crate::prelude::*;
+use jiff01::civil::{Date, DateTime, Time};
+use jiff01::{Timestamp, Zoned};
+
+#[derive(JsonSchema, Serialize, Deserialize)]
+struct JiffTypes {
+    date_time_ts: Timestamp,
+    date_time_zoned: Zoned,
+    naive_date: Date,
+    naive_date_time: DateTime,
+    naive_time: Time,
+}
+
+#[test]
+fn jiff() {
+    test!(JiffTypes).assert_snapshot();
+
+    test!(Timestamp)
+        .assert_allows_ser_roundtrip_default()
+        .assert_matches_de_roundtrip(arbitrary_values());
+
+    // test!(Zoned)
+    //     .assert_allows_ser_roundtrip_default()
+    //     .assert_matches_de_roundtrip(arbitrary_values());
```

Comment:
Failed locally when running test:

```
failures:

---- jiff::jiff stdout ----
thread 'jiff::jiff' panicked at schemars/tests/integration/test_helper.rs:193:13:
serialize schema should allow serialized value: "1970-01-01T00:00:00+00:00[UTC]"
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace


failures:
```

Not sure if we need to extend the test helper but I'd include the minimal work code. Other improvements can be made as follow-ups.

Reply:
Fixed in https://github.com/GREsau/schemars/pull/364/commits/3855539148343d22fc6e187c1aa8a4ef5119d92d

1. Use a custom format because jiff's `Zoned` default format is not "date-time". I'd like to know where is the definition of "date-time" though.
2. Enable default features for jiff in tests, it pulls the tzdb so that the test can work properly.

Reply:
Found https://json-schema.org/draft/2020-12/draft-bhutton-json-schema-validation-01#name-dates-times-and-duration

While jiff's Zoned default serialize format is not fully RFC-3339 but instead "A hybrid format derived from RFC 3339, RFC 9557 and ISO 8601" - https://docs.rs/jiff/latest/jiff/fmt/temporal/index.html

Reply:
To be clear here, _specifically_, `jiff::Zoned` serializes to a fully compatible RFC 9557 timestamp. [RFC 9557](https://datatracker.ietf.org/doc/rfc9557/) is an update to RFC 3339.

Additionally, a `jiff::Timestamp` serializes to a fully compatible RFC 3339 timestamp.

Reply:
> Additionally, a jiff::Timestamp serializes to a fully compatible RFC 3339 timestamp.

Yes. I tag `jiff::Timestamp` with `format: 'date-time'` and it works well.

> jiff::Zoned serializes to a fully compatible RFC 9557 timestamp

Good to know. IIUC before JSON Schema provide a builtin format for it, or extend the `date-time` format to accept RFC 9557, we should be fine with the workaround for a custom format. Out of curiosity, is the timestamp format in RFC-3339 a subset of RFC-9557's?

Reply:
I think that makes sense. I'm not super familiar with JSON Schema though. Where is the custom format you're using documented? Like I would kind of expect to see a mention of RFC 9557 somewhere.

> Out of curiosity, is the timestamp format in RFC-3339 a subset of RFC-9557's?

I would say yes. Although whether RFC 9557's clarification on the meaning of `Z` and `-00:00` is considered a superset of RFC 3339 is perhaps up for debate, but it's a very small point. In terms of the actual format supported, yes, RFC 9557 is a superset. Basically, RFC 9557 adds and defines the semantics for annotations to be appended to an RFC 3339 timestamp.

Reply:
> Where is the custom format you're using documented?

It's customized ("zoned-date-time") in this lib rather than documented elsewhere. I filed https://github.com/json-schema-org/json-schema-spec/issues/1572 to see if the JSON Schema working group agrees to extend the definition of "date-time" format to accept timestamp strings in RFC 9557 format.